frontend/pages/admin/IntegrationsPage/cards/Mdm/Mdm.tsx

```diff
@@ -0,0 +1,292 @@
+    day: "numeric",
+  }).format(dateString);
+};
+
+const Mdm = (): JSX.Element => {
+  const { isPremiumTier, isMdmEnabled } = useContext(AppContext);
+  const { renderFlash } = useContext(NotificationContext);
+
+  const [showRequestModal, setShowRequestModal] = useState(false);
+
+  const {
+    data: mdmApple,
+    isLoading: isLoadingMdmApple,
+    error: errorMdmApple,
+  } = useQuery<IMdmApple, Error, IMdmApple>(
+    ["mdmAppleAPI"],
+    () => mdmAppleAPI.loadAll(),
+    {
+      enabled: isMdmEnabled && isPremiumTier,
+      staleTime: 5000,
+    }
+  );
+
+  const {
+    data: mdmAppleBm,
+    isLoading: isLoadingMdmAppleBm,
+    error: errorMdmAppleBm,
+  } = useQuery<IMdmAppleBm, Error, IMdmAppleBm>(
+    ["mdmAppleBmAPI"],
+    () => mdmAppleBmAPI.loadAll(),
+    {
+      enabled: isMdmEnabled && isPremiumTier,
+      staleTime: 5000,
+    }
+  );
+
+  // MDM TODO: Test manually after backend is merged
+  const {
+    data: keys,
+    error: fetchKeysError,
+    isFetching: isFetchingKeys,
+  } = useQuery<string, Error>(["keys"], () => mdmAppleBmAPI.loadKeys(), {
+    enabled: !!isMdmEnabled && isPremiumTier,
+    refetchOnWindowFocus: false,
+  });
+
+  const toggleRequestModal = () => {
+    setShowRequestModal(!showRequestModal);
+  };
+
+  const onDownloadKeys = (evt: React.MouseEvent) => {
+    evt.preventDefault();
+
+    // MDM TODO: Confirm error flash message
+    if (isFetchingKeys || fetchKeysError) {
+      renderFlash(
+        "error",
+        "Your MDM business manager keys could not be downloaded. Please try again."
+      );
+      return false;
+    }
+
+    if (keys) {
+      // MDM TODO: Validate keys like we validate certificates?
+      // if (keys && isValidKeys(keys)) {
+      const filename = "fleet.pem";
+      const file = new global.window.File([keys], filename, {
+        type: "application/x-pem-file",
+      });
+
+      FileSaver.saveAs(file);
+    } else {
+      renderFlash(
+        "error",
+        "Your MDM business manager keys could not be downloaded. Please TODO ACTION."
+      );
+    }
+    return false;
+  };
+
+  const renderMdmAppleSection = () => {
+    if (errorMdmApple) {
+      return <DataError />;
+    }
+
+    if (!mdmApple) {
+      return (
+        <>
+          <div className={`${baseClass}__section-description`}>
+            Connect Fleet to Apple Push Certificates Portal to change settings
+            and install software on your macOS hosts.
+          </div>
+          <div className={`${baseClass}__section-instructions`}>
+            <p>
+              1. Request a certificate signing request (CSR) and key for Apple
+              Push Notification Service (APNs) and a certificate and key for
+              Simple Certificate Enrollment Protocol (SCEP).
+            </p>
+            <Button onClick={toggleRequestModal} variant="brand">
+              Request
+            </Button>
+            <p>2. Go to your email to download to download your CSR.</p>
```

Comment:
little typo there

Reply:
Whoops, copy and pasted these without reading them >.<